api/src/main/java/org/infinispan/api/reactive/Cache.java

```diff
@@ -46,90 +47,93 @@
     * @param value
     * @return Void
     */
-   CompletionStage<Void> save(K key, V value);
+   Uni<V> put(K key, V value);
+
+   /**
+    *
+    * @param key
+    * @param value
+    * @return
+    */
+   Uni<Void> set(K key, V value);
 
    /**
     * Delete the key
     *
     * @param key
-    * @return Void
+    * @return
     */
-   CompletionStage<Void> delete(K key);
+   Uni<Boolean> remove(K key);
 
    /**
     * Retrieve all keys
     *
     * @return Publisher
     */
-   Publisher<K> keys();
+   Multi<K> keys();
 
    /**
     * Retrieve all entries
     *
     * @return
     */
-   Publisher<? extends Map.Entry<K, V>> entries();
+   Multi<? extends Map.Entry<K, V>> entries();
 
    /**
     * Save many from a Publisher
     *
     * @param pairs
     * @return Void
     */
-   Publisher<WriteResult<K>> saveMany(Publisher<Map.Entry<K, V>> pairs);
+   Multi<WriteResult<K>> put(Multi<Map.Entry<K, V>> pairs);
 
    /**
     * Estimate the size of the store
     *
     * @return Long, estimated size
     */
-   CompletionStage<Long> estimateSize();
+   Uni<Long> estimateSize();
 
    /**
     * Clear the store. If a concurrent operation puts data in the store the clear might not properly work
     *
     * @return Void
     */
-   CompletionStage<Void> clear();
+   Uni<Void> clear();
 
    /**
     * Executes the query and returns a reactive streams Publisher with the results
     *
-    * @param ickleQuery query String
-    * @return Publisher reactive streams
+    * @param query query String
+    * @return
     */
-   Publisher<KeyValueEntry<K, V>> find(String ickleQuery);
+   <R> Multi<R> find(String query);
 
    /**
     * Find by QueryRequest.
     *
-    * @param queryRequest
+    * @param query
     * @return
     */
-   Publisher<KeyValueEntry<K, V>> find(QueryRequest queryRequest);
-
-   /**
-    * Executes the query and returns a reactive streams Publisher with the results
-    *
-    * @param ickleQuery query String
-    * @return Publisher reactive streams
-    */
-   Publisher<KeyValueEntry<K, V>> findContinuously(String ickleQuery);
+   <R> Query<R> query(String query);
 
    /**
     * Executes the query and returns a reactive streams Publisher with the results
     *
-    * @param queryRequest
+    * @param query query String
     * @return Publisher reactive streams
     */
-   <T> Publisher<KeyValueEntry<K, T>> findContinuously(QueryRequest queryRequest);
+   <R> Multi<R> findContinuously(String query);
```

Comment:
Don't continuous queries have the notion of being removed and added to a query? Maybe this should return something similar to a `Multi<KeyValueQueryEvent<K,V>>` that designates if it is was a removal or not

Reply:
Fixed.